src/settings/UserListView.ts

```diff
@@ -24,70 +23,69 @@ import {isUpdateForTypeRef} from "../api/main/EventController"
+
+	constructor(
+		private readonly settingsView: SettingsView
+	) {
+		this.listId = new LazyLoaded(async () => {
+			const customer = await locator.entityClient.load(CustomerTypeRef, logins.getUserController().user.customer!)
+			return customer.userGroups
 		})
 		this.list = new List({
 			rowHeight: size.list_row_height,
-			fetch: (startId, count) => {
-				if (startId === GENERATED_MAX_ID) {
-					return this._loadAdmins().then(() => {
-						return this._listId.getAsync().then(listId => {
-							return locator.entityClient.loadAll(GroupInfoTypeRef, listId).then(allUserGroupInfos => {
-								// we have to set loadedCompletely to make sure that fetch is never called again and also that new users are inserted into the list, even at the end
-								this._setLoadedCompletely()
+			fetch: async (startId, count) => {
+				if (startId !== GENERATED_MAX_ID) {
+					throw new Error("fetch user group infos called for specific start id")
+				}
+				await this.loadAdmins()
+				const listId = await this.listId.getAsync()
+				const allUserGroupInfos = await locator.entityClient.loadAll(GroupInfoTypeRef, listId)
 
-								// we return all users because we have already loaded all users and the scroll bar shall have the complete size.
-								if (logins.getUserController().isGlobalAdmin()) {
-									return allUserGroupInfos
-								} else {
-									let localAdminGroupIds = logins
-										.getUserController()
-										.getLocalAdminGroupMemberships()
-										.map(gm => gm.group)
-									return allUserGroupInfos.filter((gi: GroupInfo) => gi.localAdmin && localAdminGroupIds.indexOf(gi.localAdmin) !== -1)
-								}
-							})
-						})
-					})
+				// we have to set loadedCompletely to make sure that fetch is never called again and also that new users are inserted into the list, even at the end
+				this.setLoadedCompletely()
+
+				// we return all users because we have already loaded all users and the scroll bar shall have the complete size.
+				if (logins.getUserController().isGlobalAdmin()) {
+					return allUserGroupInfos
 				} else {
-					throw new Error("fetch user group infos called for specific start id")
+					let localAdminGroupIds = logins
+						.getUserController()
+						.getLocalAdminGroupMemberships()
+						.map(gm => gm.group)
+					return allUserGroupInfos.filter((gi: GroupInfo) => gi.localAdmin && localAdminGroupIds.indexOf(gi.localAdmin) !== -1)
 				}
 			},
-			loadSingle: elementId => {
-				return this._listId.getAsync().then(listId => {
-					return locator.entityClient
-								  .load<GroupInfo>(GroupInfoTypeRef, [listId, elementId])
-								  .catch(
-									  ofClass(NotFoundError, e => {
-										  // we return null if the entity does not exist
-										  return null
-									  }),
-								  )
-				})
+			loadSingle: async elementId => {
+				const listId = await this.listId.getAsync()
+				try {
+					return await locator.entityClient.load<GroupInfo>(GroupInfoTypeRef, [listId, elementId])
```

Comment:
I still think that `.catch(ofClass(...))` might look much better here but it's your call